include/circt/Dialect/Moore/MooreTypes.td

```diff
@@ -52,6 +52,46 @@ def EventType : MooreTypeDef<"Event", [], "moore::UnpackedType"> {
 }
 
 
+//===----------------------------------------------------------------------===//
+// Packed Types
+//===----------------------------------------------------------------------===//
+
+def IntType : MooreTypeDef<"Int", [], "moore::PackedType"> {
+  let typeName = "moore.int";
```

Comment:
I think one should not use this directly, but the mnemonic instead. However, since you have custom parsers and printers, you shouldn't have to set this at all.